src/dbos-runtime/config.ts

```diff
@@ -337,3 +348,158 @@ function isValidDBname(dbName: string): boolean {
   }
   return validator.matches(dbName, '^[a-z0-9_]+$');
 }
+
+/*
+ This function takes a DBOSConfig and ensure that 'public' fields take precedence over 'internal' fields.
+ It assumes that the DBOSConfig was passed programmatically and thus does not need to consider CLI options.
+
+ - Application Name: check there is no inconsistency between the provided name and the one in dbos-config.yaml, if any
+ - Database configuration: Ignore provided poolConfig and reconstructs it from the database_url field and constructPoolConfig()
+*/
+export function translatePublicDBOSconfig(config: DBOSConfig): [DBOSConfig, DBOSRuntimeConfig] {
```

Comment:
Seems like translatePublicDBOSconfig needs a forceConsole parameter to correctly configure `dbosConfig.telemetry.logs.forceConsole`